### ISSUE TITLE
Compiler crash trying to port FastParse

### ISSUE THREAD

**Issue by lihaoyi** [scala-native/scala-native#434]
No idea what's wrong, but it should not crash.

Repro instructions provided below
```
lihaoyi test$ git clone https://github.com/scala-native-ports/sourcecode -b scalaxhack
Cloning into 'sourcecode'...
remote: Counting objects: 395, done.
remote: Compressing objects: 100% (21/21), done.
remote: Total 395 (delta 7), reused 0 (delta 0), pack-reused 360
Receiving objects: 100% (395/395), 321.09 KiB | 285.00 KiB/s, done.
Resolving deltas: 100% (165/165), done.
lihaoyi test$ cd sourcecode/
lihaoyi sourcecode$ sbt ++2.11.8 sourcecodeNative/publishLocal
[info] Loading project definition from /Users/lihaoyi/Dropbox/Workspace/test/sourcecode/project
[info] Updating {file:/Users/lihaoyi/Dropbox/Workspace/test/sourcecode/project/}sourcecode-build...
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[info] Done updating.
[info] Set current project to sourcecode (in build file:/Users/lihaoyi/Dropbox/Workspace/test/sourcecode/)
[info] Setting version to 2.11.8
[info] Reapplying settings...
[info] Set current project to sourcecode (in build file:/Users/lihaoyi/Dropbox/Workspace/test/sourcecode/)
[info] Updating {file:/Users/lihaoyi/Dropbox/Workspace/test/sourcecode/}sourcecodeNative...
[info] Packaging /Users/lihaoyi/Dropbox/Workspace/test/sourcecode/sourcecode/native/target/scala-2.11/sourcecode_native0.1.0-SNAPSHOT_2.11-0.1.4-SNAPSHOT-sources.jar ...
[info] Done packaging.
[info] Wrote /Users/lihaoyi/Dropbox/Workspace/test/sourcecode/sourcecode/native/target/scala-2.11/sourcecode_native0.1.0-SNAPSHOT_2.11-0.1.4-SNAPSHOT.pom
[info] Resolving jline#jline;2.12.1 ...
[info] Done updating.
[info] :: delivering :: com.lihaoyi#sourcecode_native0.1.0-SNAPSHOT_2.11;0.1.4-SNAPSHOT :: 0.1.4-SNAPSHOT :: integration :: Sat Dec 10 15:21:32 GMT 2016
[info] 	delivering ivy file to /Users/lihaoyi/Dropbox/Workspace/test/sourcecode/sourcecode/native/target/scala-2.11/ivy-0.1.4-SNAPSHOT.xml
[info] Main Scala API documentation to /Users/lihaoyi/Dropbox/Workspace/test/sourcecode/sourcecode/native/target/scala-2.11/api...
[info] Compiling 2 Scala sources to /Users/lihaoyi/Dropbox/Workspace/test/sourcecode/sourcecode/native/target/scala-2.11/classes...
[warn] dropping dependency on node with no phase object: mixin
[warn] there were three feature warnings; re-run with -feature for details
model contains 39 documentable templates
[warn] one warning found
[info] Main Scala API documentation successful.
[info] Packaging /Users/lihaoyi/Dropbox/Workspace/test/sourcecode/sourcecode/native/target/scala-2.11/sourcecode_native0.1.0-SNAPSHOT_2.11-0.1.4-SNAPSHOT-javadoc.jar ...
[info] Done packaging.
[warn] there were two deprecation warnings; re-run with -deprecation for details
[warn] there were three feature warnings; re-run with -feature for details
[warn] two warnings found
[info] Packaging /Users/lihaoyi/Dropbox/Workspace/test/sourcecode/sourcecode/native/target/scala-2.11/sourcecode_native0.1.0-SNAPSHOT_2.11-0.1.4-SNAPSHOT.jar ...
[info] Done packaging.
[info] 	published sourcecode_native0.1.0-SNAPSHOT_2.11 to /Users/lihaoyi/.ivy2/local/com.lihaoyi/sourcecode_native0.1.0-SNAPSHOT_2.11/0.1.4-SNAPSHOT/poms/sourcecode_native0.1.0-SNAPSHOT_2.11.pom
[info] 	published sourcecode_native0.1.0-SNAPSHOT_2.11 to /Users/lihaoyi/.ivy2/local/com.lihaoyi/sourcecode_native0.1.0-SNAPSHOT_2.11/0.1.4-SNAPSHOT/jars/sourcecode_native0.1.0-SNAPSHOT_2.11.jar
[info] 	published sourcecode_native0.1.0-SNAPSHOT_2.11 to /Users/lihaoyi/.ivy2/local/com.lihaoyi/sourcecode_native0.1.0-SNAPSHOT_2.11/0.1.4-SNAPSHOT/srcs/sourcecode_native0.1.0-SNAPSHOT_2.11-sources.jar
[info] 	published sourcecode_native0.1.0-SNAPSHOT_2.11 to /Users/lihaoyi/.ivy2/local/com.lihaoyi/sourcecode_native0.1.0-SNAPSHOT_2.11/0.1.4-SNAPSHOT/docs/sourcecode_native0.1.0-SNAPSHOT_2.11-javadoc.jar
[info] 	published ivy to /Users/lihaoyi/.ivy2/local/com.lihaoyi/sourcecode_native0.1.0-SNAPSHOT_2.11/0.1.4-SNAPSHOT/ivys/ivy.xml
[success] Total time: 8 s, completed 10 Dec, 2016 3:21:40 PM
lihaoyi sourcecode$ cd ..
lihaoyi test$ git clone https://github.com/scala-native-ports/fastparse -b scalaxhack
Cloning into 'fastparse'...
remote: Counting objects: 7730, done.
remote: Compressing objects: 100% (2239/2239), done.
remote: Total 7730 (delta 2867), reused 7730 (delta 2867), pack-reused 0
Receiving objects: 100% (7730/7730), 3.50 MiB | 815.00 KiB/s, done.
Resolving deltas: 100% (2867/2867), done.
lihaoyi test$ cd fastparse/
lihaoyi fastparse$ sbt fastparseNative/run
[info] Loading project definition from /Users/lihaoyi/Dropbox/Workspace/test/fastparse/project
[info] Updating {file:/Users/lihaoyi/Dropbox/Workspace/test/fastparse/project/}fastparse-build...
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[info] Done updating.
[info] Compiling 1 Scala source to /Users/lihaoyi/Dropbox/Workspace/test/fastparse/project/target/scala-2.10/sbt-0.13/classes...
[info] Set current project to fastparse (in build file:/Users/lihaoyi/Dropbox/Workspace/test/fastparse/)
[info] Updating {file:/Users/lihaoyi/Dropbox/Workspace/test/fastparse/}utilsNative...
[info] Resolving jline#jline;2.12.1 ...
[info] Done updating.
[info] Updating {file:/Users/lihaoyi/Dropbox/Workspace/test/fastparse/}fastparseNative...
[info] Resolving jline#jline;2.12.1 ...
[info] Done updating.
[info] Compiling 6 Scala sources to /Users/lihaoyi/Dropbox/Workspace/test/fastparse/utils/native/target/scala-2.11/classes...
[warn] there was one deprecation warning; re-run with -deprecation for details
[warn] one warning found
[info] Because /Users/lihaoyi/Dropbox/Workspace/test/fastparse/utils/shared/src/main/scala/fastparse/utils/Utils.scala contains a macro definition, the following dependencies are invalidated unconditionally:
[info] 	/Users/lihaoyi/Dropbox/Workspace/test/fastparse/utils/shared/src/main/scala/fastparse/utils/ParserInput.scala
[info] Compiling 16 Scala sources to /Users/lihaoyi/Dropbox/Workspace/test/fastparse/fastparse/native/target/scala-2.11/classes...
[warn] /Users/lihaoyi/Dropbox/Workspace/test/fastparse/fastparse/shared/src/main/scala/fastparse/TestMain.scala:8: a pure expression does nothing in statement position; you may be omitting necessary parentheses
[warn]     fastparse.all.implicitReprOps
[warn]                   ^
[warn] /Users/lihaoyi/Dropbox/Workspace/test/fastparse/fastparse/shared/src/main/scala/fastparse/parsers/Combinators.scala:347: abstract type R in type pattern fastparse.parsers.Combinators.Sequence[R,R,R,Elem,Repr] is unchecked since it is eliminated by erasure
[warn]           case p: Sequence[R, R, R, Elem, Repr] =>
[warn]                   ^
[warn] there were 6 feature warnings; re-run with -feature for details
[warn] three warnings found
scala.scalanative.util.package$UnsupportedException: Is(F(64),Local(Local(src,1),Ptr)) (class scala.scalanative.nir.Op$Is)
	at scala.scalanative.util.package$.unsupported(package.scala:19)
	at scala.scalanative.codegen.CodeGen$Impl$$anonfun$27.apply(CodeGen.scala:482)
	at scala.scalanative.codegen.CodeGen$Impl$$anonfun$27.apply(CodeGen.scala:434)
	at scala.scalanative.util.Show$$anon$1.apply(Show.scala:67)
	at scala.scalanative.util.Show$.toResult(Show.scala:81)
	at scala.scalanative.codegen.CodeGen$Impl.showLet(CodeGen.scala:430)
	at scala.scalanative.codegen.CodeGen$Impl.showInst(CodeGen.scala:299)
	at scala.scalanative.codegen.CodeGen$Impl$$anonfun$showInsts$1.apply(CodeGen.scala:288)
	at scala.scalanative.codegen.CodeGen$Impl$$anonfun$showInsts$1.apply(CodeGen.scala:288)
	at scala.collection.mutable.UnrolledBuffer$Unrolled.foreach(UnrolledBuffer.scala:224)
	at scala.collection.mutable.UnrolledBuffer.foreach(UnrolledBuffer.scala:130)
	at scala.scalanative.codegen.CodeGen$Impl.showInsts(CodeGen.scala:288)
	at scala.scalanative.codegen.CodeGen$Impl.showBlock(CodeGen.scala:167)
	at scala.scalanative.codegen.CodeGen$Impl$$anonfun$9.apply(CodeGen.scala:151)
	at scala.scalanative.codegen.CodeGen$Impl$$anonfun$9.apply(CodeGen.scala:149)
	at scala.scalanative.optimizer.analysis.ControlFlow$Graph$$anonfun$map$1.apply(ControlFlow.scala:50)
	at scala.scalanative.optimizer.analysis.ControlFlow$Graph$$anonfun$map$1.apply(ControlFlow.scala:49)
	at scala.scalanative.optimizer.analysis.ControlFlow$Graph.foreach(ControlFlow.scala:41)
	at scala.scalanative.optimizer.analysis.ControlFlow$Graph.map(ControlFlow.scala:49)
	at scala.scalanative.codegen.CodeGen$Impl.showFunctionDefn(CodeGen.scala:149)
	at scala.scalanative.codegen.CodeGen$Impl$$anonfun$4.apply(CodeGen.scala:77)
	at scala.scalanative.codegen.CodeGen$Impl$$anonfun$4.apply(CodeGen.scala:69)
	at scala.scalanative.util.Show$$anon$1.apply(Show.scala:67)
	at scala.scalanative.util.Show$.toResult(Show.scala:81)
	at scala.scalanative.codegen.CodeGen$Impl$$anonfun$2$$anonfun$apply$1.apply(CodeGen.scala:66)
	at scala.scalanative.codegen.CodeGen$Impl$$anonfun$2$$anonfun$apply$1.apply(CodeGen.scala:66)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:47)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:244)
	at scala.collection.AbstractTraversable.map(Traversable.scala:105)
	at scala.scalanative.codegen.CodeGen$Impl$$anonfun$2.apply(CodeGen.scala:66)
	at scala.scalanative.codegen.CodeGen$Impl$$anonfun$2.apply(CodeGen.scala:56)
	at scala.scalanative.util.Show$$anon$1.apply(Show.scala:67)
	at scala.scalanative.codegen.CodeGen$Impl.gen(CodeGen.scala:54)
	at scala.scalanative.tools.package$$anonfun$codegen$1.apply(package.scala:64)
	at scala.scalanative.tools.package$$anonfun$codegen$1.apply(package.scala:63)
	at scala.scalanative.io.package$.withScratchBuffer(package.scala:13)
	at scala.scalanative.tools.package$.codegen(package.scala:63)
	at scala.scalanative.sbtplugin.ScalaNativePluginInternal$.scala$scalanative$sbtplugin$ScalaNativePluginInternal$$compileNir(ScalaNativePluginInternal.scala:104)
	at scala.scalanative.sbtplugin.ScalaNativePluginInternal$$anonfun$unscopedSettings$10$$anonfun$apply$4$$anonfun$16.apply(ScalaNativePluginInternal.scala:296)
	at scala.scalanative.sbtplugin.ScalaNativePluginInternal$$anonfun$unscopedSettings$10$$anonfun$apply$4$$anonfun$16.apply(ScalaNativePluginInternal.scala:288)
	at sbt.FileFunction$$anonfun$cached$1.apply(Tracked.scala:253)
	at sbt.FileFunction$$anonfun$cached$1.apply(Tracked.scala:253)
	at sbt.FileFunction$$anonfun$cached$2$$anonfun$apply$3$$anonfun$apply$4.apply(Tracked.scala:267)
	at sbt.FileFunction$$anonfun$cached$2$$anonfun$apply$3$$anonfun$apply$4.apply(Tracked.scala:263)
	at sbt.Difference.apply(Tracked.scala:224)
	at sbt.Difference.apply(Tracked.scala:206)
	at sbt.FileFunction$$anonfun$cached$2$$anonfun$apply$3.apply(Tracked.scala:263)
	at sbt.FileFunction$$anonfun$cached$2$$anonfun$apply$3.apply(Tracked.scala:262)
	at sbt.Difference.apply(Tracked.scala:224)
	at sbt.Difference.apply(Tracked.scala:200)
	at sbt.FileFunction$$anonfun$cached$2.apply(Tracked.scala:262)
	at sbt.FileFunction$$anonfun$cached$2.apply(Tracked.scala:260)
	at scala.scalanative.sbtplugin.ScalaNativePluginInternal$$anonfun$unscopedSettings$10$$anonfun$apply$4.apply(ScalaNativePluginInternal.scala:311)
	at scala.scalanative.sbtplugin.ScalaNativePluginInternal$$anonfun$unscopedSettings$10$$anonfun$apply$4.apply(ScalaNativePluginInternal.scala:242)
	at scala.scalanative.util.Scope$.apply(Scope.scala:30)
	at scala.scalanative.sbtplugin.ScalaNativePluginInternal$$anonfun$unscopedSettings$10.apply(ScalaNativePluginInternal.scala:242)
	at scala.scalanative.sbtplugin.ScalaNativePluginInternal$$anonfun$unscopedSettings$10.apply(ScalaNativePluginInternal.scala:242)
	at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
	at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
	at sbt.std.Transform$$anon$4.work(System.scala:63)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
	at sbt.Execute.work(Execute.scala:237)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
	at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[error] (fastparseNative/*:nativeLink) scala.scalanative.util.package$UnsupportedException: Is(F(64),Local(Local(src,1),Ptr)) (class scala.scalanative.nir.Op$Is)
[error] Total time: 22 s, completed 10 Dec, 2016 3:23:11 PM
```

**Comment by densh** [comment]
Thanks! I have a hunch that this one and https://github.com/scala-native/scala-native/issues/326 are closely related.

**Comment by densh** [comment]
Fixed in 6a10bc8
